in_memory_device_dao.go

```diff
@@ -16,3 +16,15 @@ func (db *inMemoryDeviceDAO) Save(device Device) (Device, error) {
 
 	return device, nil
 }
+
+func (db *inMemoryDeviceDAO) GetByID(id int) (*Device, error) {
+	db.mu.Lock()
+	defer db.mu.Unlock()
```

Comment:
Do you need to lock on read? I think we can safely read without the lock (as long as we don't remove from the list, and we are not doing that ;) )